pyiron_contrib/atomistic/atomicrex/structure_list.py

```diff
@@ -0,0 +1,278 @@
+                struct.to_hdf(hdf=hdf_s_lst, group=f"structure_{k}")
+            
+    def from_hdf(self, hdf, group="structure_list"):
+        with hdf.open(group) as hdf_s_lst:
+            for g in sorted(hdf_s_lst.list_groups()):
+                structure = Atoms()
+                structure.from_hdf(hdf, group_name = g)
+                self.append(structure)
+        
+    def to_ase_db(self, db):
+        print("NOT TESTED FUNCTION")
+        for struct in self.structures:
+            struct = pyiron_to_ase(struct)
+            db.write(atoms=struct, forces=self.forces, key_value_pairs=struct.info)
+            
+    def from_ase_db(self, db):
+        for row in db.select():
+            kvp = row.key_value_pairs
+            if "pyiron_id" in kvp.keys():
+                p_id = kvp["pyiron_id"]
+            else:
+                raise KeyError("pyiron_id has to be supplied as key in row to transform to structure list")
+            i = row.data            
+            i.update(kvp)
+            a = row.toatoms()
+            structure = ase_to_pyiron(a)
+            structure.info = i
+            print(i)
+            self[f"pyiron_id{p_id}"] = structure
+
+
+### This is probably useless like this because forces can't be passed.
+def structure_to_xml_element(structure):
+    pbc = ET.Element("pbc")
+    pbc.set("x", f"{structure.pbc[0]}".lower())
+    pbc.set("y", f"{structure.pbc[1]}".lower())
+    pbc.set("z", f"{structure.pbc[2]}".lower())
+    
+    c = structure.cell
+    cell = ET.Element("cell")
+    for i in range(3):
+        a = ET.SubElement(cell, f"a{i+1}")
+        a.set("x", f"{c[i][0]}")
+        a.set("y", f"{c[i][1]}")
+        a.set("z", f"{c[i][2]}")
+    
+    atoms = ET.SubElement(cell, "atoms")
+    for at in structure:
+        a = ET.SubElement(atoms, "atom")
+        a.set("type", at.symbol)
+        a.set("x", f"{at.a}")
+        a.set("y", f"{at.b}")
+        a.set("z", f"{at.c}")
+        a.set("reduced", "false")
+    
+    return pbc, cell
+
+
+class ARStructure(object):
+    #def __new__(cls, *args, **kwargs):
+    #    instance = super().__new__(cls)
+    #    object.__setattr__(instance, "structure", Atoms())
+
+    def __init__(self, structure=None, fit_properties=None, identifier=None, relative_weight=1, clamp=True,):
+        self.structure = structure
+        self.fit_properties = fit_properties
+        self.identifier = identifier
+        self.relative_weight = relative_weight
+        self.clamp = clamp
+    
+    @property
+    def structure(self):
+        return self._structure
+    
+    @structure.setter
+    def structure(self, structure):
+        if not isinstance(structure, Atoms):
+            raise ValueError("structure has to be a pyiron atoms instance")
+        self._structure = structure
+    
+    @property
+    def fit_properties(self):
+        return self._fit_properties
+    
+    @fit_properties.setter
+    def fit_properties(self, fit_properties=ARFitPropertyList()):
```

Comment:
This creates a single `ARFitPropertyList` that will be the default value for all calls to this method.  This is probably not what you want.